### PR TITLE
Add marshalized receipts in storer

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Added in receipts storer as a value a marshalized batch which contains all the marshalized receipts in clear and not only their hashes

Testing steps:

a) Prerequisites

b) Test steps

Start a normal testnet

c) Expected results

We should have another storage/folder/db created in: .../Epoch_X/Shard_Y/Receipts which holds cumulative receipt hash as a key and all the marshalized receipts list marshalized as a value, for each committed block

d) Actual results

Test process is in pending

### CODE REVIEW

File: process/coordinator/process.go [hunk 1]
```diff
@@ -952,39 +952,38 @@ func (tc *transactionCoordinator) CreateReceiptsHash() ([]byte, error) {
...
 
-		mb := interProc.GetCreatedInShardMiniBlock()
-		if mb == nil {
+		miniBlock := interProc.GetCreatedInShardMiniBlock()
+		if miniBlock == nil {
 			continue
 		}
 
-		mbHash, err := core.CalculateHash(tc.marshalizer, tc.hasher, mb)
+		marshalizedMiniBlock, err := tc.marshalizer.Marshal(miniBlock)
```
Comment (sasurobert) [inline process/coordinator/process.go:969]: this is a list of marshalled miniblocks. make sure that these things are not saved 2 times.

**Review by sasurobert — COMMENTED** [review]
the description does not match the implementation - here you are saving receipt hash - all miniblocks in clear (but those will contain hashes one by one). It is good - maybe we should ensure that receiptshashes are not saved 2 times in storage.

File: process/coordinator/process.go [hunk 2]
```diff
@@ -952,39 +952,38 @@ func (tc *transactionCoordinator) CreateReceiptsHash() ([]byte, error) {
...
 	}
 
-	if len(receiptsHashes) == 0 {
+	if len(receiptsBatch.Data) == 0 {
 		return nil, nil
 	}
 
-	receiptsBatch := &batch.Batch{Data: receiptsHashes}
-	marshalizedReceiptsHashes, err := tc.marshalizer.Marshal(receiptsBatch)
+	marshalizedReceiptsBatch, err := tc.marshalizer.Marshal(receiptsBatch)
```
Comment (bogdan-rosianu) [inline process/coordinator/process.go:981]: could return directly

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: process/coordinator/process.go (hunk 1 above)
Comment (SebastianMarian) [reply process/coordinator/process.go:969]: They are not saved anywhere as these are the ones which are deleted from applyBodyToHeader through newBody := deleteSelfReceiptsMiniBlocks(body) method

**Review by bogdan-rosianu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

File: process/coordinator/process.go (hunk 2 above)
Comment (SebastianMarian) [reply process/coordinator/process.go]: done

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**